concordium-std/src/test_infrastructure.rs

```diff
@@ -1588,6 +1648,36 @@ impl<State: Serial + DeserialWithState<TestStateApi>> TestHost<State> {
         *self.contract_balance.borrow_mut() = amount;
     }
 
+    pub fn set_self_address(&mut self, address: ContractAddress) {
```

Comment:
Document, including the precondition for the relationship with `set_missing_contract`.